tests/unit/pyrflx_test.py

```diff
@@ -1203,3 +1204,93 @@ def test_always_valid_aspect(  # pylint: disable=invalid-name
+) -> None:
+    array_message_one = array_message_package["Array_Member"]
+    array_message_one.set("Byte", 5)
+    array_message_two = array_message_package["Array_Member"]
+    array_message_two.set("Byte", 6)
+    sdu_message = array_message_package["Array_Member"]
+    sdu_message.set("Byte", 7)
+
+    message_foo_value = message_array_refinement_value
+    array_contents: Sequence[TypeValue] = [array_message_one, array_message_two]
+    message_foo_value.set("Length", 2)
+    message_foo_value.set("Array_Field", array_contents)
+    message_foo_value.set("Payload", sdu_message.bytestring)
+    assert message_foo_value.valid_message
+
+    inner_messages = message_foo_value.inner_messages()
+    assert len(inner_messages) == 3
+    assert all(isinstance(m, MessageValue) for m in inner_messages)
+    assert {m.get("Byte") for m in inner_messages} == {5, 6, 7}
+
+
+def test_get_path(icmp_message_value: MessageValue) -> None:
+    test_bytes = (
+        b"\x08\x00\xe1\x1e\x00\x11\x00\x01\x4a\xfc\x0d\x00\x00\x00\x00\x00"
+        b"\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f"
+        b"\x20\x21\x22\x23\x24\x25\x26\x27\x28\x29\x2a\x2b\x2c\x2d\x2e\x2f"
+        b"\x30\x31\x32\x33\x34\x35\x36\x37"
+    )
+    icmp_message_value.parse(test_bytes)
+    covered_links = icmp_message_value.path
+    field_pairs = [
+        ("Initial", "Tag"),
+        ("Tag", "Code_Zero"),
+        ("Code_Zero", "Checksum"),
+        ("Checksum", "Identifier"),
+        ("Identifier", "Sequence_Number"),
+        ("Sequence_Number", "Data"),
+        ("Data", "Final"),
+    ]
+
+    assert len(covered_links) == 7
+    for link, field_pair in zip(covered_links, field_pairs):
+        assert link.source.name == field_pair[0] and link.target.name == field_pair[1]
+
+
+def test_get_path_non_sequential(icmp_message_value: MessageValue) -> None:
+    test_data = (
+        b"\x47\xb4\x67\x5e\x00\x00\x00\x00"
+        b"\x4a\xfc\x0d\x00\x00\x00\x00\x00\x10\x11\x12\x13\x14\x15\x16\x17"
+        b"\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f\x20\x21\x22\x23\x24\x25\x26\x27"
+        b"\x28\x29\x2a\x2b\x2c\x2d\x2e\x2f\x30\x31\x32\x33\x34\x35\x36\x37"
+    )
+    icmp_message_value.set("Tag", "Echo_Request")
+    icmp_message_value.set("Code_Zero", 0)
+    icmp_message_value.set("Checksum", 1234)
+    icmp_message_value.set("Identifier", 5)
+    icmp_message_value.set("Sequence_Number", 1)
+    icmp_message_value.set("Data", test_data)
+    covered_links = icmp_message_value.path
+    field_pairs = [
+        ("Initial", "Tag"),
+        ("Tag", "Code_Zero"),
+        ("Code_Zero", "Checksum"),
+        ("Checksum", "Identifier"),
+        ("Identifier", "Sequence_Number"),
+        ("Sequence_Number", "Data"),
+        ("Data", "Final"),
+    ]
+    assert len(covered_links) == 7
+    for link, field_pair in zip(covered_links, field_pairs):
+        assert link.source.name == field_pair[0] and link.target.name == field_pair[1]
```

Comment:
Could be replaced by just:

```Python
assert [(l.source.name, l.target.name) for l in covered_links] == field_pairs
```

There are similar instances.

Reply:
Changed in c0a80f2

Reply:
The `assert len(covered_links) == 7` should not be needed anymore. Pytest usually shows a detailed error message in case of missing or additional elements in one of the compared list.

Reply:
Removed in c883635